Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine+State.swift

```diff
@@ -40,8 +40,12 @@ extension ExperienceStateMachine {
                 )
             case let (.endingStep(experience, currentIndex, _), .startStep(stepRef)):
                 return Transition.fromEndingStepToBeginningStep(experience, currentIndex, stepRef, traitComposer)
-            case (.endingExperience, .reset):
-                return Transition(toState: .idling)
+            case let (.endingExperience(experience, stepIndex, markComplete), .reset):
+                var sideEffect: SideEffect?
+                if markComplete || stepIndex == experience.stepIndices.last {
```

Comment:
one minor suggestion would be to extract this little bit of logic here and shared on line 85 of ExperienceStateObserver to a single spot, if possible -- checking if an experience was completed or not.

Reply:
Updated in f31db7dd227160260b6846c71535755dfe03fc2c because you're right that should have the logic a single place, even if it's not quite as elegant as I'd hoped.

It's tricky to do elegantly because the analytics observer has a `Result<ExperienceStateMachine.State>` versus this just being an `ExperienceStateMachine.State` so I can't nicely extend either one of those to encapsulate this logic.

Basically, if I do
```swift
extension ExperienceStateMachine.State {
        var isExperienceCompleted: Bool { ... }
}
```
then I don't have access to the `State` var in the AnalyticsObserver because it's stuck in the `Result` type, so I need extensions on both.